src/main/java/ch/njol/skript/effects/EffApplyBoneMeal.java

```diff
@@ -0,0 +1,79 @@
+/**
+ *   This file is part of Skript.
+ *
+ *  Skript is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Skript is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright Peter Güttinger, SkriptLang team and contributors
+ */
+package ch.njol.skript.effects;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.lang.Effect;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.util.Kleenean;
+import org.bukkit.block.Block;
+import org.bukkit.block.BlockFace;
+import org.bukkit.event.Event;
+import org.eclipse.jdt.annotation.Nullable;
+
+@Name("Apply Bone Meal")
+@Description("Applies bone meal to a crop, sapling, or composter")
+@Examples("apply 3 bone meal to event-block")
+@Since("INSERT VERSION")
+public class EffApplyBoneMeal extends Effect {
+
+	static {
+		Skript.registerEffect(EffApplyBoneMeal.class, "apply [%-number%] bone[ ]meal[s] to %blocks%");
+	}
+
+	private Expression<Block> blocks;
+	@Nullable
+	private Expression<Number> amount;
+
+	@Override
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		amount = (Expression<Number>) exprs[0];
+		blocks = (Expression<Block>) exprs[1];
+		return true;
+	}
+
+	@Override
+	protected void execute(Event event) {
+		int times = 1;
+		if (amount != null) {
+			Number n = amount.getSingle(event);
+			if (n == null)
+				return;
+			times = n.intValue();
+		}
+		for (Block block : blocks.getArray(event)) {
+			for (int i = 0; i < times; i++) {
+				block.applyBoneMeal(BlockFace.UP);
```

Comment:
I don't know in Minecraft if applying bone meal in any other direction may be of use? Maybe the syntax could support optionally a direction and default to UP?

Reply:
That would require adding a classinfo for blockface. I can't think of any use for optional blockfaces and it seems useless to me.

Reply:
It would be a %direction%

Reply:
Please see https://discord.com/channels/135877399391764480/836220422223036467/1138867292931891260.